src/heif.imageio/CMakeLists.txt

```diff
@@ -3,6 +3,33 @@
 # https://github.com/OpenImageIO/oiio
 
 if (Libheif_FOUND)
+   if (LINKSTATIC)
+    if (WIN32)
+        set (_static_suffixes .lib .a)
+    else ()
+        set (_static_suffixes .a)
+    endif ()
+
+    set (_static_libraries_found 0)
+    foreach(_libeheif_library IN LISTS LIBHEIF_LIBRARIES)
+       get_filename_component(_ext ${_libeheif_library} LAST_EXT)
+       list (FIND _static_suffixes ${_ext} _index)
+       if (${_index} GREATER -1)
+         MATH(EXPR _static_libraries_found "${static_libraries_found}+1")
+       endif()
+    endforeach()
+
+    if (${_static_libraries_found} GREATER 0)
+     message("${ColorRed}")
+     message(STATUS "Looks like you are linking against some static libheif libraries")
+     message(STATUS "libheif is a LGPL-licensed library")
+     message(STATUS "When linking statically against a LGPL-licensed library your entire project needs to be under LGPL")
```

Comment:
Let's omit the prescriptive legal advice and go with a simpler informational message that leaves the problem on their doorstep:

```
You are linking OpenImageIO against a static version of libheif, which is LGPL
licensed. If you intend to redistribute this build of OpenImageIO, we recommend
that you review the libheif license terms, or you may wish to switch to using a
dynamically-linked libheif.
```

I think this gives them the information they need to point them in the right direction, and avoids letting them stumble unaware into possible trouble, but without trying to tell them exactly what they can and can't do (which, as non-lawyers, we should not be doing and could be wrong about).

A slightly different message more accurately covers the ffmpeg situation:

```
You are linking OpenImageIO against a static version of FFmpeg, which may have
LGPL and possibly GPL licensed components (depending on exactly how your copy
of FFmpeg was built). If you intend to redistribute this build of OpenImageIO,
we recommend that you review the FFmpeg build flags and licensing terms.
```